File: public/credits.js
/** Human project credits (keep in sync with installer/src-tauri/src/credits.rs). */
window.SB_CREDITS = {
  creator: { name: "Rahil Pirani", github: "rahilp" },
  maintainers: [
    { name: "Vincenzo Fabiano" },
    { name: "Aneesh Grover", github: "Aneesh-382005" },
    { name: "Mike Stanley", github: "mikestanley00" },
    { name: "Mochammad Fadhlan Al-Ghiffari", github: "MFA-G" },
    { name: "Phillip Smith", github: "phillipadsmith" },
    { name: "Robert Brandin", github: "tumes" },
    { name: "oudouusa", github: "oudouusa" },
  ],
};

window.renderAboutCredits = function renderAboutCredits() {
  const root = document.getElementById("about-credits");
  if (!root) return;

  const { creator, maintainers } = window.SB_CREDITS;
  const personLine = (p) => {
    if (p.github) {
      const url = `https://github.com/${p.github}`;
      return `${p.name} (<a href="${url}" target="_blank" rel="noopener noreferrer">@${p.github}</a>)`;
    }
    return p.name;
  };

  root.innerHTML = `
    <div class="about-credits-label">${escHtml(t('menu.about'))}</div>
    <p class="about-credits-line"><span class="about-credits-kicker">${escHtml(t('menu.createdBy'))}</span> ${personLine(creator)}</p>
    <p class="about-credits-kicker">${escHtml(t('menu.maintainers'))}</p>
    <ul class="about-credits-list">
      ${maintainers.map((p) => `<li>${personLine(p)}</li>`).join("")}
    </ul>
  `;
};

File: installer/src-tauri/src/credits.rs
//! Project credits for the native About dialog and in-app About section.
//! Human contributors from git history; bots and automated agents are excluded.

use crate::i18n::{self, Key, Locale};

pub struct Person {
    pub name: &'static str,
    pub github: Option<&'static str>,
}

/// Original author of Second Brain.
pub const CREATOR: Person = Person {
    name: "Rahil Pirani",
    github: Some("rahilp"),
};

/// Everyone with commits in this repository except bots / AI automation accounts.
pub const MAINTAINERS: &[Person] = &[
    Person {
        name: "Vincenzo Fabiano",
        github: None,
    },
    Person {
        name: "Aneesh Grover",
        github: Some("Aneesh-382005"),
    },
    Person {
        name: "Mike Stanley",
        github: Some("mikestanley00"),
    },
    Person {
        name: "Mochammad Fadhlan Al-Ghiffari",
        github: Some("MFA-G"),
    },
    Person {
        name: "Phillip Smith",
        github: Some("phillipadsmith"),
    },
    Person {
        name: "Robert Brandin",
        github: Some("tumes"),
    },
    Person {
        name: "oudouusa",
        github: Some("oudouusa"),
    },
];

fn line(person: &Person) -> String {
    match person.github {
        Some(handle) => format!("{} (@{handle})", person.name),
        None => person.name.to_string(),
    }
}

/// macOS About panel — multiline credits field.
pub fn credits_text(locale: Locale) -> String {
    let mut out = format!(
        "{} {}\n\n{}\n",
        i18n::t(locale, Key::CreditsCreatedBy),
        line(&CREATOR),
        i18n::t(locale, Key::CreditsMaintainersLabel),
    );
    for person in MAINTAINERS {
        out.push_str(&format!("• {}\n", line(person)));
    }
    out.trim_end().to_string()
}

/// Windows / Linux About — author list (creator first, then maintainers).
pub fn author_names() -> Vec<String> {
    let mut names = vec![line(&CREATOR)];
    names.extend(MAINTAINERS.iter().map(line));
    names
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn credits_include_creator_and_maintainers() {
        let text = credits_text(Locale::En);
        assert!(text.contains("Rahil Pirani"));
        assert!(text.contains("Vincenzo Fabiano"));
        assert!(text.contains("Created by"));
        assert!(!text.contains("dependabot"));
        assert!(!text.contains("bot]"));
    }

    #[test]
    fn credits_localized_in_italian() {
        let text = credits_text(Locale::It);
        assert!(text.contains("Creato da"));
        assert!(text.contains("Manutentori:"));
    }

    #[test]
    fn authors_list_starts_with_creator() {
        let authors = author_names();
        assert!(authors[0].contains("Rahil"));
        assert_eq!(authors.len(), 1 + MAINTAINERS.len());
    }
}
